wpilibc/src/main/native/cpp/PneumaticHub.cpp

```diff
@@ -56,12 +57,36 @@ class PneumaticHub::DataStore {
         std::shared_ptr<DataStore>{this, wpi::NullDeleter<DataStore>()};
 
     auto version = m_moduleObject.GetVersion();
-    if (version.FirmwareMajor > 0 && version.FirmwareMajor < 22) {
-      throw FRC_MakeError(
-          err::AssertionFailure,
-          "The Pneumatic Hub has firmware version {}.{}.{}, and must be "
-          "updated to version 2022.0.0 or later using the REV Hardware Client",
-          version.FirmwareMajor, version.FirmwareMinor, version.FirmwareFix);
+
+    if (version.FirmwareMajor > 0) {
+      // Write PH firmware version to roboRIO
```

Comment:
Please wrap this in a `if (IsReal())` so it doesn't try to run in simulation.  Likewise Java should check `RobotBase.isReal()`